### PR TITLE
NMS-13197: Compile OpenNMS with JDK11

### PR DESCRIPTION
**Pull request by j-white**
JIRA: https://issues.opennms.org/browse/NMS-13197

This patch includes build system and library upgrades to support compiling JDK11.
The tree can no longer be compiled or run with JDK8 once this is merged.

### CODE REVIEW

**Comment by RangerRick** [comment]
Having not had a chance to look through the patches yet, let me say just 2 words: hell, and yeah.

File: bin/functions.pl [hunk 1]
```diff
@@ -214,7 +216,7 @@ END
 	$ENV{'PATH'}      = File::Spec->catfile($JAVA_HOME, 'bin') . $PATHSEP . $ENV{'PATH'};
 
         my ($shortversion) = get_version_from_java(File::Spec->catfile($JAVA_HOME, 'bin', 'java'));
-        if ($shortversion >= 9) {
+        if ($shortversion >= 9 && $shortversion < 11) {
```
Comment (RangerRick) [inline bin/functions.pl:219]: Doesn't this basically only set the variable if JDK is 9 or 10?  What's the reason for keeping this if minversion=11 anyway?

**Review by RangerRick — APPROVED** [review]
Looks pretty good, nothing jumps out at me.

File: bin/functions.pl (hunk 1 above)
Comment (j-white) [reply bin/functions.pl]: Good point. Removed.

**Review by indigo423 — APPROVED** [review]
LGTM! 👍  Happy you haven't upgraded the docker container from 20.10.1. They [broke](https://github.com/moby/moby/issues/41858) IPv6 in 20.10.2 and might get us in trouble with IPv6 tests. I haven't seen anything suspicious to me in the PR.

**Pull request merged**